lib/ansible/parsing/vault/__init__.py

```diff
@@ -344,23 +343,21 @@ def __repr__(self):
 
 
 class ScriptVaultSecret(FileVaultSecret):
-
-    @staticmethod
-    def read_file(filename, loader):
-        if not loader.is_executable(filename):
+    def read_file(self, filename):
+        if not self.loader.is_executable(filename):
             raise AnsibleVaultError("The vault password script %s was not executable" % filename)
 
         try:
             # STDERR not captured to make it easier for users to prompt for input in their scripts
             p = subprocess.Popen(filename, stdout=subprocess.PIPE)
         except OSError as e:
-            msg_format = "Problem running vault password script %s (%s)."
-            "If this is not a script, remove the executable bit from the file."
-            msg = msg_format % (' '.join(filename), e)
+            msg_format = "Problem running vault password script %s (%s)." \
+                " If this is not a script, remove the executable bit from the file."
+            msg = msg_format % (filename, e)
 
             raise AnsibleError(msg)
 
-        stdout, stderr = p.communicate()
+        stdout, _ = p.communicate()
```

Comment:
we've avoided using  _ as a 'unused var'  in case we decide to use i18n .. which also has special uses for _